src/main/java/domain/piece/jump/Pho.java

```diff
@@ -0,0 +1,73 @@
+package domain.piece.jump;
+
+import static domain.piece.PieceType.PHO;
+
+import domain.Coordinate;
+import domain.board.ReadableBoard;
+import domain.piece.Country;
+import domain.piece.Piece;
+import domain.piece.movement.Movement;
+import domain.piece.movement.Movements;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Pho extends Piece {
+
+    private final Movements movements = new Movements(List.of(
+            Movement.UP, Movement.DOWN, Movement.RIGHT, Movement.LEFT));
+
+    public Pho(Country country) {
+        super(country, PHO);
+    }
+
+    @Override
+    public List<Coordinate> findAvailablePaths(Coordinate from, ReadableBoard readableBoard) {
+        movements.addMovementIfInGung(from);
+
+        List<Coordinate> availablePositions = new ArrayList<>();
+
+        for (Movement movement : movements.getMovements()) {
+            Coordinate next = from.move(movement);
+            boolean isJumped = false;
+
+            while (movement.isDiagonal() ? next.isInGungBoundary() : next.isInBoundary()) {
+                StepDecision step = decide(readableBoard, next, isJumped);
+
+                if (step.shouldAdd()) {
+                    availablePositions.add(next);
+                }
+                if (step.shouldStop()) {
+                    break;
+                }
+                if (step.shouldJump()) {
+                    isJumped = true;
+                }
+
+                next = next.move(movement);
+            }
+        }
+        return availablePositions;
+    }
+
+    public StepDecision decide(ReadableBoard readableBoard, Coordinate to, boolean isJumped) {
+        if (!isJumped) {
+            if (readableBoard.hasPiece(to)) {
+                if (readableBoard.findPieceTypeByCoordinate(to) == PHO) {
+                    return StepDecision.stop();
+                }
+                return StepDecision.of(false, false, true);
+            }
+            return StepDecision.skip();
+        }
```

Comment:
아래와 같이 하면 depth 를 줄일 수 있을 것 같아요!

```suggestion
    if (!isJumped) {
        if (!readableBoard.hasPiece(to)) {
            return StepDecision.skip();
        }
        if (readableBoard.findPieceTypeByCoordinate(to) == PHO) {
            return StepDecision.stop();
        }
        return StepDecision.of(false, false, true);
    }
```

Reply:
감사합니다 🥲
skip일 경우 얼리 리턴을 통해 분기를 줄일 수 있었군요!